endpoints.go

```diff
@@ -33,3 +42,312 @@ func showHealth(w http.ResponseWriter, r *http.Request) {
+            return
+        }
+    }
+
+    queryString := "SELECT survey_ref, short_name, long_name, legal_basis, survey_mode FROM surveyv2.survey" + sb.String()
+
+    rows, err := db.Query(queryString)
+
+    if err != nil {
+        http.Error(w, "get survey query failed", http.StatusInternalServerError)
+        return
+    }
+
+    var listOfSurveys = []models.Survey{}
+
+    for rows.Next(){
+        survey := models.Survey{}
+
+        rows.Scan(
+            &survey.Reference,
+            &survey.ShortName,
+            &survey.LongName,
+            &survey.LegalBasis,
+            &survey.SurveyMode,
+        )
+
+        listOfSurveys = append(listOfSurveys, survey)
+    }
+
+    if len(listOfSurveys) == 0 {
+        re := models.NewRESTError("404", "Survey not found")
+        data, err := json.Marshal(re)
+        if err != nil {
+            http.Error(w, "Error marshaling NewRestError JSON", http.StatusInternalServerError)
+            return
+        }
+
+        w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+        w.WriteHeader(http.StatusNotFound)
+        w.Write(data)
+
+        return
+    }
+
+    data, err := json.Marshal(listOfSurveys)
+    if err != nil {
+        http.Error(w, "Failed to marshal survey JSON", http.StatusInternalServerError)
+        return
+    }
+
+    logger.Logger.Info("Successfully retrieved survey")
+    w.Header().Set("Content-Type", "application/json; charset=UTF-8")
+    w.WriteHeader(http.StatusOK)
+    w.Write(data)
+}
+
+//Create survey based on JSON request
+func postSurvey (w http.ResponseWriter, r *http.Request) {
+
+    if db == nil {
+        w.WriteHeader(http.StatusInternalServerError)
+        errorString := models.Error{
+            Error: "Database connection could not be found",
+        }
+        json.NewEncoder(w).Encode(errorString)
+        return
+    }
+
+    //replace with schemav2 when done
+
+    stmt, err := db.Prepare("INSERT INTO surveyv2.survey VALUES($1, $2, $3, $4, $5)")
+
+    if err != nil {
+        http.Error(w, "SQL statement not prepared", http.StatusInternalServerError)
+        return
+    }
+
+    body, err := ioutil.ReadAll(r.Body)
+    if err != nil {
+        http.Error(w, "Couldn't read message body", http.StatusInternalServerError)
+        return
+    }
+
+    var survey models.Survey
+
+    err = json.Unmarshal(body, &survey)
+
+    if err != nil {
+        http.Error(w, "Error unmarshalling JSON", http.StatusBadRequest)
+        return
+    }
+
+    // Generate a UUID to uniquely identify the new survey
+    surveyRef, err := uuid.NewV4()
+    if err != nil {
+        http.Error(w, "Error generating random uuid", http.StatusInternalServerError)
+        return
+    }
+
+    survey.Reference = surveyRef.String()
+
+    _, err = stmt.Exec(survey.Reference, survey.ShortName, survey.LongName, survey.LegalBasis, survey.SurveyMode)
+    if err != nil {
+        http.Error(w, "SQL statement error", http.StatusInternalServerError)
```

Comment:
```suggestion
        http.Error(w, "SQL statement error:" + err.Error(), http.StatusInternalServerError)
```

We should be flagging up what's wrong - for example, we might have duplicate survey references, the consuming service should know about that so they can fix it.